base/src/com/google/idea/blaze/base/lang/buildfile/psi/TargetExpression.java

```diff
@@ -34,6 +38,28 @@ protected void acceptVisitor(BuildElementVisitor visitor) {
     visitor.visitTargetExpression(this);
   }
 
+  @NotNull
+  @Override
+  public PsiElement getNavigationElement() {
+    PsiElement referencedElement = getReferencedElement();
+    if(referencedElement != null) {
+      return referencedElement.getNavigationElement();
+    }
+    return super.getNavigationElement();
+  }
+
+  @Nullable
+  @Override
+  public PsiElement getReferencedElement() {
```

Comment:
this is likely wrong -- it should break rename refactoring and reference references I think -- will fix after feedback on #249